app/forms/add_to_order_form.rb

```diff
@@ -230,15 +230,33 @@ def merge_order
                        created_by: created_by.id,
                        cross_core_project: @order_project,
                      )
-                   else
+                    else
                      target_order
-                   end
+                    end
+  end
+
+  def find_or_create_product_order
```

Comment:
Add a code comment to explain why are we finding first?
"Create the new order if the project doesn't have a cross core order for the selected facility yet"